### DISCUSSION TITLE
Is it possible to train a GNN to learn the relationship between input and output graphs with differing structures?

### DISCUSSION THREAD

**Discussion by jagandecapri** [pyg-team/pytorch_geometric#7810] [Q&A]
Hi,

In the problem I'm working, given a batch, each input graph maps to an output graph. The input and output graph does not have the same node and edge structures. The input and output graph can be of varying sizes. 

Is it possible to train a GNN to learn the relationship between the input and output graphs?

**Comment by rusty1s** [comment]
Interesting problem. I would assume that something like `GAE` is not directly applicable to that use-case due to the difference in node sizes. Alternatively, you may find success in revisiting some literature on graph generation, but it is sadly not my main field of expertise.
